research/design-doc/sparse_checkout_asishkumar.md

```diff
@@ -0,0 +1,68 @@
+# KPM sparse checkout
+
+**Author**: Asish Kumar
+
+## Abstract
+
+`kpm` manages third-party libraries through Git repositories, requiring a `kcl.mod` file at the root directory. It treats the entire Git repository as a single `kcl` package, which is inefficient for monorepos containing multiple `kcl` packages. Often, a `kcl` project depends on just one package within a monorepo, but `kpm` downloads the entire repository. Therefore, `kpm` needs to allow adding a subdirectory of a Git repository as a dependency, enabling it to download only the necessary parts and improve performance.
+
+## User Interface
+
+I will add a new flag called `--subdir` in `kcl mod add` command.  This flag will specify the path to the desired subdirectory within the Git repository. Below is the syntax for the enhanced kpm add command:
+
+```
+kcl mod add --git <git-repo-url> --subdir <subdir> 
```

Comment:
This is a small question. I would like to ask why this method needs to be separated into two parameters.

```
kcl mod add --git <git-repo-url>/<subdir>
```

Will this cause some problems ?

Reply:
Thanks. That's a much better approach. now i don't even have to add checks to avoid subdir download operation in other source than git. I have updated the design doc.